### PR TITLE
I added a progress bar/value meter thing and a 16 bit register

### PR DESCRIPTION
**Pull request by alexwaeseperlman**

### CODE REVIEW

**Comment by alexwaeseperlman** [comment]
I can't build it though cause I'm not set up

File: src/main.c [hunk 1]
```diff
@@ -15,38 +15,42 @@ typedef unsigned short uint16_t;
...
 //The speed of the patterns
 //Param: an integer in the range 0 - 255. The total speed is made of a high byte and a low byte.
 //CMD_SPEEDHIGH changes the high byte, while CMD_SPEEDLOW changes the low byte.
-#define CMD_SPEEDHIGH	0x07
-#define CMD_SPEEDLOW	0x08
+#define CMD_SPEEDHIGH 0x07
+#define CMD_SPEEDLOW 0x08
+// Set the value of the register with least significant and most significant bits
+#define REG_SETVALUE_MSB 0x09
+#define REG_SETVALUE_LSB 0x0A
```
Comment (tylertian123) [inline src/main.c]: I suggest following the naming convention and naming these `CMD_REG_MSB` or something similar.

File: src/main.c [hunk 2]
```diff
@@ -15,38 +15,42 @@ typedef unsigned short uint16_t;
...
 //Resets the BeautifulRobot. All properties (e.g. brightness, color) will be reset back to the default.
 //Param: none
-#define CMD_RESET		0x09
+#define CMD_RESET 0xff
```
Comment (tylertian123) [inline src/main.c]: `CMD_RESET` cannot be 0xFF, as the value 0xFF is already used for the sync byte which marks the termination of a command. Even though the command is only checked when the buffer is full, the value 0xFF as a command ID can have unintended consequences and mess with syncing.

File: src/main.c [hunk 3]
```diff
@@ -58,17 +62,21 @@ volatile bit direction = 0;
...
 volatile uint8_t speedLow = 0x00;
 
-xdata volatile RGBColor colors[80] = { 0 };
+volatile uint16_t reg = 0;
+
+xdata volatile RGBColor colors[80] = {0};
 xdata volatile unsigned short time = 0;
-void processCmd(uint16_t cmdBuf) {
+void processCmd(uint16_t cmdBuf)
+{
```
Comment (tylertian123) [inline src/main.c]: I really don't get why you took the time to change every single pair of curly braces to a different style......

File: src/main.c [hunk 4]
```diff
@@ -85,94 +93,115 @@ void processCmd(uint16_t cmdBuf) {
...
+		LED_COUNT = param;
+		break;
 	case CMD_SPEEDHIGH:
 		speedHigh = param;
 		break;
 	case CMD_SPEEDLOW:
 		speedLow = param;
 		break;
+	case REG_SETVALUE_LSB:
+		reg = param | (reg >> 8) << 8;
```
Comment (tylertian123) [inline src/main.c]: Instead of doing `reg >> 8 << 8` just do `reg & 0xFF00`, it executes much faster and is more clear.

File: src/main.c [hunk 5]
```diff
@@ -85,94 +93,115 @@ void processCmd(uint16_t cmdBuf) {
...
+		break;
+	case REG_SETVALUE_MSB:
+		reg = param << 8 | (reg << 8) >> 8;
+		break;
 	case CMD_RESET:
 		//Cast 0x0000 into a function pointer and call it to reset everything
-		((void (code *) (void)) 0x0000) ();
+		((void(code *)(void))0x0000)();
```
Comment (tylertian123) [inline src/main.c]: What but why did you delete all the spaces....

**Review by tylertian123 — CHANGES_REQUESTED** [review]
I mean, I like the idea of a progress bar mode, but eh, tbh there're a lot of things that you didn't need to do (and maybe shouldn't have done).

File: src/main.c (hunk 4 above)
Comment (tylertian123) [reply src/main.c]: In the same way, change every `reg << 8 >> 8` to a `reg & 0x00FF`.

File: src/main.c (hunk 1 above)
Comment (alexwaeseperlman) [reply src/main.c]: K thats a good idea. Idk why I wanted 16 bits lol.

File: src/main.c (hunk 4 above)
Comment (alexwaeseperlman) [reply src/main.c]: Smort

**Comment by alexwaeseperlman** [comment]
I just remembered why the FASTABS is faster than normal abs. It doesn't use any if statements which means it doesn't branch so its way easier for the processor.

**Review by tylertian123 — APPROVED** [review]

**Pull request merged**